src/Controllers/UserController.js

```diff
@@ -60,11 +60,18 @@ export class UserController extends AdaptableController {
       updateFields._email_verify_token_expires_at = {__op: 'Delete'};
     }
 
-    return this.config.database.update('_User', query, updateFields).then((document) => {
-      if (!document) {
-        throw undefined;
-      }
-      return Promise.resolve(document);
+    var self = this;
+    var checkIfAlreadyVerified = new RestQuery(self.config, Auth.master(self.config), '_User', {username: username, emailVerified: true});
```

Comment:
This is redundant, you already have `aUser.emailVerified` in resendVerificationEmail, the only place that should be checked is in there.  Additionally you are now imposing extra responsibility on verifyEmail that shouldn't be part of its responsibility.  Whoever is calling this should check it if it wants.